Sources/UberAuth/AuthManager.swift

```diff
@@ -0,0 +1,44 @@
+//
+//  Copyright © Uber Technologies, Inc. All rights reserved.
+//
+
+
+import Foundation
+import UIKit
+
+public typealias AuthCompletion = (Result<Client, UberAuthError>) -> ()
+
+/// @mockable
+protocol AuthManaging {
+    
+    func login(context: AuthContext, completion: @escaping AuthCompletion)
+    
+    func handle(_ url: URL) -> Bool
+}
+
+final class AuthManager: AuthManaging {
+    
+    private var currentContext: AuthContext?
+    
+    init() {}
+    
+    // MARK: LoginManaging
+    
+    func login(context: AuthContext = .init(),
+               completion: @escaping AuthCompletion) {
+        context.authProvider.execute(
```

Comment:
Looks like this manager is just a wrapper for `context.authProvider`. That said, do we really need it?

Reply:
As discussed. I merged AuthManager and UberAuth since the purposes are similar right now. If they diverge in the future I will split them again.